common/src/main/java/io/crate/common/collections/Maps.java

```diff
@@ -92,9 +93,23 @@ public static Object getByPath(Map<String, Object> value, List<String> path) {
     /**
      * Inserts a value into source under the given key+path
      */
-    public static void mergeInto(Map<String, Object> source, String key, List<String> path, Object value) {
+    public static void mergeInto(Map<String, Object> source,
+                                 String key,
+                                 List<String> path,
+                                 Object value) {
+        mergeInto(source, key, path, value, Map::put);
+    }
+
+    /**
+     * Inserts a value into source under the given key+path
+     */
+    public static void mergeInto(Map<String, Object> source,
+                                 String key,
+                                 List<String> path,
+                                 Object value,
+                                 TriConsumer<Map<String, Object>, String, Object> writer) {
```

Comment:
Using a interface from `log4j` for this feels a bit weird. Would it be worth creating a interface in our own package?

Reply:
yes wasn't sure if its worth, will do.